src/lib-voyager.tsx

```diff
@@ -66,6 +66,63 @@ class Voyager {
   }
 
   /**
+   * Sets the entire voyager application state. This is useful for restoring
+   * the state of the application to a previosly saved state.
+   *
+   * @param state A StateBase object with the following keys
+   *
+   * @param state.config
+   * @param state.dataset
+   * @param state.shelf
+   * @param state.result
+   *
+   * @memberof Voyager
+   */
+  public setApplicationState(state: Readonly<StateBase>): void {
+    this.data = undefined;
+    this.config = undefined;
+
+    this.renderFromState(state);
+  }
+
+  /**
+   *
+   * Gets the current application state.
+   *
+   * @returns {Readonly<StateBase>}
+   *
+   * @memberof Voyager
+   */
+  public getApplicationState(): Readonly<StateBase> {
+    return this.store.getState().present;
+  }
+
+  /**
+   * Subscribe to state changes.
+   *
+   * This is useful for taking state snapshots to persist and later restore.
+   *
+   * @param {Function} onChange callback that takes a single state parameter.
+   * @returns {Function} unsubscribe, call this function to remove this listener.
+   *
+   * @memberof Voyager
+   */
+  public onStateChange(onChange: Function): Function {
+    let currentState: any;
+
+    const handleChange = () => {
+      const nextState = this.store.getState();
+      if (nextState !== currentState) {
```

Comment:
It is quite interesting that you have to check the state one more time here, but I guess that's fine.